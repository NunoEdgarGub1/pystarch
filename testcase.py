import testimport

globe = 1

class MyClass(object):
    class_attr = 1
    def class_method(self, a):
        return a

    def another_method(self, b):
        return [2]


def my_function(sam, maxi):
    x = True
    alice = 1
    bob = 2
    bob = 2 + sam
    globe = 3
    return alice + bob


def func2():
    if 4 == my_function():
        return 1
    if 'a' == my_function():
        return 2

def func():
    alice = 1
    if alice == 2:
        return []
    alice = 'a'
    return alice


def another_function():
    alice = 'hey'
    if alice == [1]:
        return 5
    if alice is None:
        return 1
    if alice == 'hello':
        return 6
    alice += ''
    alice += 2
    return None

a,b,c = [1,2,3]
a,b,c = 'abc'
a = [1,2,3]
a[:2]
c = b * 4
my_function()
instance = MyClass()
x = instance.class_attr
y = instance.another_method('a')

z = testimport.imported
with open('side_effect', 'w') as f:
    f.write('uh oh')
