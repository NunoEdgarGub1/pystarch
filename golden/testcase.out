testcases/testcase.py:6 undefined "x"
testcases/testcase.py:6 undefined "x"
testcases/testcase.py:8 undefined "object"
testcases/testcase.py:22 reassignment "bob = ..."
testcases/testcase.py:23 reassignment "globe = ..."
testcases/testcase.py:30 inconsitent-types "Eq" (Str, Num)
testcases/testcase.py:37 type-change "alice = ..." (Num -> Str)
testcases/testcase.py:37 reassignment "alice = ..."
testcases/testcase.py:38 type-change "Return" (List -> Str)
testcases/testcase.py:43 inconsitent-types "Eq" (Str, List)
testcases/testcase.py:45 inconsitent-types "Is" (Str, None)
testcases/testcase.py:49 reassignment "alice = ..."
testcases/testcase.py:50 type-change "alice = ..." (Str -> Num)
testcases/testcase.py:50 reassignment "alice = ..."
testcases/testcase.py:51 maybe-type "Return" (Num -> None)
testcases/testcase.py:54 assign-type "Str"
testcases/testcase.py:54 reassignment "a"
testcases/testcase.py:54 reassignment "b"
testcases/testcase.py:54 reassignment "c"
testcases/testcase.py:55 type-change "a = ..." (Any -> List)
testcases/testcase.py:55 reassignment "a = ..."
testcases/testcase.py:57 type-change "c = ..." (Any -> Num)
testcases/testcase.py:57 reassignment "c = ..."
testcases/testcase.py:65 type-change "i = ..." (Any -> Num)
testcases/testcase.py:65 reassignment "i = ..."
